public/app/features/alerting/unified/components/rule-viewer/__mocks__/server.ts

```diff
@@ -19,39 +19,25 @@ const folderAccess = {
   [AccessControlAction.AlertingRuleDelete]: true,
 };
 
-export function createMockGrafanaServer() {
-  const server = setupServer();
-
-  mockFolderAccess(server, folderAccess);
-  mockAlertmanagerChoiceResponse(server, alertmanagerChoiceMockedResponse);
-  mockGrafanaIncidentPluginSettings(server);
+export function createMockGrafanaServer(...handlers: RequestHandler[]) {
+  const folderHandler = mockFolderAccess(folderAccess);
+  const amChoiceHandler = mockAlertmanagerChoiceResponseHandler(alertmanagerChoiceMockedResponse);
+  const server = setupServer(folderHandler, amChoiceHandler, ...handlers);
 
   return server;
 }
 
 // this endpoint is used to determine of we have edit / delete permissions for the Grafana managed alert rule
 // a user must alsso have permissions for the folder (namespace) in which the alert rule is stored
-function mockFolderAccess(server: SetupServer, accessControl: Partial<Record<AccessControlAction, boolean>>) {
-  server.use(
-    http.get('/api/folders/:uid', ({ request }) => {
-      const url = new URL(request.url);
-      const uid = url.searchParams.get('uid');
-
-      return HttpResponse.json({
-        title: 'My Folder',
-        uid,
-        accessControl,
-      });
-    })
-  );
-
-  return server;
-}
-
-function mockGrafanaIncidentPluginSettings(server: SetupServer) {
-  server.use(
-    http.get('/api/plugins/grafana-incident-app/settings', () => {
-      return HttpResponse.json({});
-    })
-  );
+function mockFolderAccess(accessControl: Partial<Record<AccessControlAction, boolean>>) {
```

Comment:
We really need to consolidate the mock server approach, as we have `mockFolderApi` in `app/features/alerting/unified/mockApi.ts` as well 😬

Reply:
We do 😄